RIOT-OS/RIOT
# shell_commands: gnrc_netif: add support for getting/setting PHY modes & options

# Contribution description

I'm getting afraid that if #12128 gets too big, nobody will review it, so I factored out the part that makes changes to common code.

This patch adds support for setting PHY modes as specified in 802.15.4 as well as those added in IEEE 802.15.4g-2012.

If the driver implements support for it, this adds commands to configure

 - Chip/s & rate mode for O-QPSK
 - Modulation / Coding Scheme and Option for OFDM
 - Modulation Index, Modulation Order, Symbol Rate and Forward Error Correction for FSK

Modulation Index is represented as fractions of 64 to ease implementations.

Some options are also applicable for `at86rf2xx` and `mrf24j40` (probably others too, I just didn't look), they could also add more configurable for e.g. BPSK options.
I will add try to add implementations for some existing drivers to this PR.

### Testing procedure

```
ifconfig 7 set phy_mode FSK
ifconfig 7 set modulation_index 1/2
ifconfig 7 set symbol_rate 150
ifconfig 7 set forward_error_correction NRNSC
```

Should do nothing unless the driver supports it.

Or on `samr30-xpro` you should now be able to to the following:
```
ifconfig 7 set rate_mode 2
ifconfig 7 set chip_rate 400
```

You should get
```
 Iface  7  HWaddr: 34:82  Channel: 0  Page: 2  NID: 0x23 PHY: O-QPSK 
            chip rate: 600 rate mode: 2 (legacy) 
           Long HWaddr: 79:7A:34:2F:9D:73:B4:82
```

### Issues/PRs references
#12128 is the first driver to make use of this interface
#12185 implements the interface for `at86rf2xx`
